tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc

```diff
@@ -557,6 +556,63 @@ struct FoldXferReadOfXferWriterPattern
   }
 };
 
+/// convert:
+///   %0 = vector.transfer_read %0, %cst : vector<8x12xf32>
+///   %1 = arith.select %pred, %0, %vector_cst : vector<8x12xf32>
+///   %2 = vector.transfer_write %1, %0[...]
+///   %3 = vector.transfer_read %0[...], %cst : vector<8x12xf32>
+///   use(%3)
+/// to:
+///   %0 = vector.transfer_read %0, %cst : vector<8x12xf32>
+///   %1 = arith.select %pred, %0, %vector_cst : vector<8x12xf32>
+///   use(%1)
+struct FoldXferReadOfXferWriterWithSelectPattern
```

Comment:
nit: this simplification looks usual, doesn't mlir have a built-in one? :P

Reply:
No, not yet.